doc/rtd/explanation/boot.rst

```diff
@@ -3,23 +3,23 @@
 Boot stages
 ***********
 
-To be able to provide the functionality that it does, ``cloud-init`` must be
-integrated into the boot in a fairly controlled way. There are five
-stages to boot:
+There are five stages to boot:
 
-1. Generator
+1. Detect
 2. Local
 3. Network
 4. Config
 5. Final
 
-.. _boot-Generator:
+.. _boot-Detect:
 
-Generator
-=========
-When booting under ``systemd``, a generator will run that determines if
-cloud-init.target should be included in the boot goals. ``ds-identify``
-runs at this stage.
+Detect
```

Comment:
Since we've removed the mention of systemd, should we mention that this stage may not be present for every installation of cloud-init?

Reply:
Good question.

I think that the list of supported distros that don't have ds-identify integration is dwindling. At this point I think it is just Alpine, Gentoo, and OpenBSD. From some conversations in IRC, I think Alpine will gain support for it in the future (will require `ds-identify` changes to some flags for busybox tooling compatibility), and I suspect that the OpenBSD work shouldn't be too hard either. I'm happy to do the Gentoo bits if that's the last remaining distro.

I'd like to see this become an expected cloud-init behavior for future distro additions, but I'm fine with documenting how it is for now - up to you.

Reply:
Actually I just added it for now, can remove the statement when it is no longer true.

Reply:
OpenBSD has basic ds-id support, detecting discs doesn't work yet.

OpenBSD's init also doesn't support running any of our stages under it, because they don't remain running, so it gets really confused.

The strategy right now is to tell people to enable it via init.local(8).
See README in pkg/ dir: https://codeberg.org/meena/openbsd-ports/src/branch/add/cloud-init/sysutils/cloud-init
(technically, this is still init, and your statement(s) below are still correct, it's not the way people are used to from the average daemon)